js/adapt-languagePicker.js

```diff
@@ -3,12 +3,14 @@ import LanguagePickerView from './languagePickerView';
 import LanguagePickerNavView from './languagePickerNavView';
 import LanguagePickerModel from './languagePickerModel';
 
-let languagePickerModel;
-
 class LanguagePicker extends Backbone.Controller {
   
   initialize() {
-    this.listenTo(Adapt, 'configModel:dataLoaded', this.onConfigLoaded);
+    this.listenTo(Adapt, {
+      'configModel:dataLoaded': this.onConfigLoaded,
+      'router:menu router:page': this.setupNavigationView,
+      'offlineStorage:ready': this.onOfflineStorageReady
```

Comment:
Apologies if I wasn't very clear – the locations of the listeners were fine – it was just that the `LanguagePicker` class should be controlling the event  i.e. `this.listenTo()`/`this.listenToOnce()`.

We don't want the handlers firing if Language Picker isn't enabled so the listener setups should be moved back to lines 29 & 35 respectively.